src/schema/sources.ts

```diff
@@ -1023,24 +1067,68 @@ export const resolvers: IResolvers<any, Context> = {
           sourcePageGenerator.nodeToCursor(page, args, node, index),
         (builder) => {
           builder.queryBuilder
-            .addSelect('count(pk.keyword) AS pkCount')
-            .innerJoin(
-              Post,
-              'p',
-              `p.sourceId = "${builder.alias}".id AND p.createdAt > :time`,
-              { time: subDays(new Date(), 90) },
-            )
+            .addSelect('count')
             .innerJoin(
-              PostKeyword,
-              'pk',
-              'pk.postId = p.id AND pk.status = :status AND pk.keyword = :tag',
-              { status: 'allow', tag: args.tag },
+              SourceTagView,
+              'stv',
+              `stv.tag = :tag AND stv.sourceId = "${builder.alias}".id`,
+              {
+                tag: args.tag,
+              },
             )
             .andWhere(filter)
-            .groupBy(`"${builder.alias}".id`)
-            .orderBy('pkCount', 'DESC')
-            .limit(page.limit)
-            .offset(page.offset);
+            .orderBy('count', 'DESC')
+            .limit(page.limit);
+          return builder;
+        },
+      );
+    },
+    similarSources: async (
+      _,
+      args: SimilarSources,
+      ctx,
+      info,
+    ): Promise<Connection<GQLSource>> => {
+      const excludedSources = args?.excludeSources
+        ? [...alwaysExcludeSources, ...args?.excludeSources]
+        : alwaysExcludeSources;
+      const filter: FindOptionsWhere<Source> = {
+        active: true,
+        private: false,
+        id: Not(In(excludedSources)),
+      };
+
+      const subQuery = await ctx.con
+        .query(`with base as (SELECT * FROM source_tag_view), s as (
+            SELECT *, row_number() over (partition by "sourceId" order by count desc) rn
+            FROM base
+        )
+        SELECT s2."sourceId"
+        FROM s s1
+        JOIN s s2 on s1.tag = s2.tag and s1."sourceId" != s2."sourceId"
+        WHERE s1."sourceId" = '${args.sourceId}' and s1.rn <= 10 and s2.rn <= 10
+        GROUP BY 1
+        ORDER BY count(*) desc
```

Comment:
Can you order by `count` column instead?

Reply:
Oh yeah 100% this was actually just a copy paste thing 🤦